utils/scarb-oracle-hint-service/src/hint_service.rs

```diff
@@ -19,26 +23,32 @@ enum OracleCheatcodeSelectorInner {
     OracleInvoke,
 }
 
-impl Default for OracleHintService {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 /// Constructors.
 impl OracleHintService {
-    /// Creates a new `OracleHintService` with all builtin protocols enabled.
-    pub fn new() -> Self {
-        let mut this = Self::bare();
-        crate::connections::add_builtin_protocols(&mut this);
-        this
+    /// Creates a new `OracleHintService` with all builtin protocols enabled and assets directory
+    /// preconfigured.
+    ///
+    /// The `executable_path` must be a path to the file that is going to be executed by the
+    /// executor.
+    ///
+    /// ## Panics
+    /// This function will panic if the executable path is not a file.
+    pub fn new(executable_path: Option<&Path>) -> Self {
+        Self {
+            connections: Connections::default(),
+            protocols: builtin_protocols(),
+            assets: executable_path
+                .map(Assets::for_executable)
```

Comment:
Isn't this too restrictive from the perspective of the executor? 🤔 
The dd said: 

> 1. Let OracleHintService be configurable on where it allows protocols to look for assets. Basically establish a PATH-like mechanism:
>     1. Have builder-like method for adding custom paths.
>         1. Executors will be obliged use this to pass path to a directory where executed sierra/casm is located. This mechanism will allow loading assets that reside next to sierra/casm files. It can only be a convention (enforced by public shaming).

Which suggests to me that we should not limit this interface to executable location only 🤔